cradle-core/src/main/java/com/exactpro/cradle/BookAndPageChecker.java

```diff
@@ -17,26 +17,34 @@
 package com.exactpro.cradle;
 
 import java.time.Instant;
-import java.util.Map;
 
 import com.exactpro.cradle.utils.CradleStorageException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 public class BookAndPageChecker
 {
-	private final Map<BookId, BookInfo> books;
+	private final static Logger logger = LoggerFactory.getLogger(BookAndPageChecker.class);
+	private final BookCache bookCache;
 	
-	public BookAndPageChecker(Map<BookId, BookInfo> books)
+	public BookAndPageChecker(BookCache bookCache)
 	{
-		this.books = books;
+		this.bookCache = bookCache;
 	}
 	
 	
 	public BookInfo getBook(BookId bookId) throws CradleStorageException
 	{
-		BookInfo result = books.get(bookId);
-		if (result == null)
-			throw new CradleStorageException("Book '"+bookId+"' is unknown");
-		return result;
+		try {
+			return bookCache.getBook(bookId);
+		} catch (CradleStorageException e) {
+			// Book was not found
+			throw e;
```

Comment:
Why catch an exception just to re-throw it?

Reply:
I think this code is outdated, please refer to this version:
https://github.com/th2-net/cradleapi/blob/316cd25d45eaeda7361b7e78321904cd8b5aa594/cradle-cassandra/src/main/java/com/exactpro/cradle/cassandra/ReadThroughBookCache.java#L59